code/game/machinery/doors/airlock.dm

```diff
@@ -1181,10 +1217,41 @@ About the new airlock wires panel:
 
 	use_power(360)	//360 W seems much more appropriate for an actuator moving an industrial door capable of crushing people
 	has_beeped = 0
-	if(arePowerSystemsOn())
-		playsound(src, close_sound_powered, 50, 1, volume_channel = VOLUME_CHANNEL_DOORS)
-	else
-		playsound(src, open_sound_unpowered, 75, 1, volume_channel = VOLUME_CHANNEL_DOORS)
+	for(var/P in player_list)
+		var/mob/M = P
+		if(!M || !M.client)
+			continue
+		var/old_sounds = M.client.is_preference_enabled(/datum/client_preference/old_door_sounds)
+		var/department_door_sounds = M.client.is_preference_enabled(/datum/client_preference/department_door_sounds)
+		var/sound
+		var/volume
+		if(old_sounds)
+			if(arePowerSystemsOn())
+				sound = legacy_close_powered
+				volume = 50
+			else
+				sound = open_sound_unpowered
+				volume = 75
+		else if(!old_sounds && department_door_sounds && src.department_close_powered) // Else, we have old sounds disabled, the door has per-department door sounds, and we have chosen to play department door sounds, use these.
+			if(arePowerSystemsOn())
+				sound = department_close_powered
+				volume = 50
+			else
+				sound = open_sound_unpowered
+				volume = 75
+		else
+			if(arePowerSystemsOn())
+				sound = close_sound_powered
+				volume = 50
+			else
+				sound = open_sound_unpowered
+				volume = 75
+
+		var/turf/T = get_turf(M)
+		var/distance = get_dist(T, get_turf(src))
+		if(distance <= world.view * 2)
+			if(T && T.z == get_z(src))

```

Comment:
Out of scope, I think (P. sure playsound doesn't do any multiz stuff) but should doors seen from above play their opening sound? Or more broadly should playsound be given a multiz-considering counterpart?